### ISSUE TITLE
fix(recipe): editing ingredient emits no events

### ISSUE THREAD

**Issue by adam-sroka** [DigitalNutritionalAssessment/gibsonify#5]
Editing an ingredient from the `IngredientPage` by using the `FormFields` or the "Save Ingredient" `FloatingActionButton` emits no states to the `RecipeBloc`, which results in the list of ingredients in the `RecipeScreen` not rebuilding when it should:

https://user-images.githubusercontent.com/45038919/141224647-8f8f8b6b-5125-4e77-88a0-f3fb9f7549b5.mp4

and

https://user-images.githubusercontent.com/45038919/141224659-eee664a3-ea35-41be-a805-ef79cda36e72.mp4

**Comment by adam-sroka** [comment]
This is being dealt with in #4 .

**Comment by adam-sroka** [comment]
@sshazril I think another thing to try is to remove the nesting of BlocBuilders — I'm just briefly going through the code and we are nesting `RecipeScreen` inside a `BlocBuilder` in `recipe_page.dart`, then `RecipeScreen` nests `RecipeForm` inside a `BlocBuilder` in its `build` method and also in `ingredient_page.dart` `IngredientPage` nests `IngredientForm` in `BlocBuilder` and `IngredientForm` also nests widgets in a `BlocBuilder` in its `build` method.

I'd look how similar widget structures are handled in Flutter BLoC examples.

**Comment by adam-sroka** [comment]
I'd also look into building without using the `dropdown_plus` Dropdowns. They are most likely not the issue, just throwing some ideas around 🤷‍♂️.

**Comment by adam-sroka** [comment]
Incidentally closed this as the issue was linked to #4. I'll reopen it and create a new draft PR for it.

**Comment by sshazril** [comment]
Resolved in commit https://github.com/DigitalNutritionalAssessment/gibsonify/commit/118743f4fb45cd60f7b6d75023ee83ef2097e9fc in https://github.com/DigitalNutritionalAssessment/gibsonify/tree/feat/recipe

**Comment by adam-sroka** [comment]
Great @sshazril 🙌, I'll have a look at it once you request review.

Just to keep track of things, It'd be better to close this once the fix has been merged to `dev` — so opening a PR and linking this issue, and when the PR gets approved and merged the issue will be closed automatically.

As in my previous comment, I opened a new draft PR (#7) for it (and an associated branch), but feel free to delete those and open a PR for the branch in which you implemented the fix.
